File: algorithms/baekjoon/1149.js
const fs = require("fs");
const rawInput = fs.readFileSync("input").toString().trim(); // fs.readFileSync(0, "utf-8").trim()

const houses = rawInput
  .split("\n")
  .slice(1)
  .map((row) => row.split(" ").map(Number));
const table = [null, houses[0]];
for (let i = 2; i <= houses.length; i++) {
  table[i] = [
    Math.min(table[i - 1][1], table[i - 1][2]) + houses[i - 1][0],
    Math.min(table[i - 1][0], table[i - 1][2]) + houses[i - 1][1],
    Math.min(table[i - 1][0], table[i - 1][1]) + houses[i - 1][2],
  ];
}

console.log(Math.min(...table[houses.length]));

File: algorithms/baekjoon/1149.py
# 제출 시 파일 입출력 부분 지우기

import sys

sys.stdin = open("input", "r")


def solution():
    N = int(input())
    prices = []
    for _ in range(N):
        prices.append(list(map(int, input().split(" "))))
    table = [[None, None, None] for _ in range(N)]
    table[0] = prices[0]
    for i in range(1, N):
        table[i][0] = min(table[i - 1][1], table[i - 1][2]) + prices[i][0]
        table[i][1] = min(table[i - 1][0], table[i - 1][2]) + prices[i][1]
        table[i][2] = min(table[i - 1][0], table[i - 1][1]) + prices[i][2]
    print(min(table[N - 1][0], table[N - 1][1], table[N - 1][2]))


solution()
